### PR TITLE
Add cloudpickle

### PR DESCRIPTION
**Pull request by jakirkham**
Simply generated the recipe using the wonderful `conda skeleton pypi`. Made a few tweaks to make it fit nicely in conda-forge.

### CODE REVIEW

**Comment by jakirkham** [comment]
@rgbkrk @ogrisel @mrocklin (and possibly others), any thoughts on the recipe as present here.

Also, would any of you like to be added as maintainers on this recipe? This will let you have commit privileges on the repo used to build this package for `conda`. Could be used to bump the version number at a release for instance or make other changes.

**Comment by jakirkham** [comment]
Passing. :)

**Comment by ogrisel** [comment]
I wouldn't mind be added as a maintainer. We need to add conda packaging instructions to the release instructions in the project repo.

File: recipes/cloudpickle/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+$PYTHON setup.py install
```
Comment (ocefpaf) [inline recipes/cloudpickle/build.sh]: We can remove this file and put that in the `build/script`.

File: recipes/cloudpickle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  version: {{ version }}
+
+source:
+  fn: cloudpickle-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/c/cloudpickle/cloudpickle-{{ version }}.tar.gz
+  md5: d6c92acdfa253a50b2ef86a1b6443e39
+
+build:
+  number: 0
+
```
Comment (ocefpaf) [inline recipes/cloudpickle/meta.yaml]: cloudpickle can use setuptools, so you can add

`script: python setup.py install --single-version-externally-managed --record record.txt`

Comment (ogrisel) [inline recipes/cloudpickle/meta.yaml]: Good remark. This makes it possible to `pip uninstall` it properly.

File: recipes/cloudpickle/build.sh (hunk 1 above)
Comment (jakirkham) [inline recipes/cloudpickle/build.sh]: Right, done.

File: recipes/cloudpickle/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/cloudpickle/meta.yaml]: Yep, forgot about this. Now added.

File: recipes/cloudpickle/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+about:
+  home: https://github.com/cloudpipe/cloudpickle
+  license: BSD License
+  summary: 'Extended pickling support for Python objects'
+
+extra:
+  recipe-maintainers:
+    - jakirkham
+    - ogrisel
```
Comment (ocefpaf) [inline recipes/cloudpickle/meta.yaml]: :+1:

**Comment by jakirkham** [comment]
Thanks @ogrisel. Added you as a maintainer.

**Comment by jakirkham** [comment]
Any more comments?

**Comment by ocefpaf** [comment]
> Any more comments?

Nope. I guess we don't need to wait for AppVeyor on this one.  Merging!

**Comment by jakirkham** [comment]
Great, thanks everyone.

**Comment by jakirkham** [comment]
Just FYI, @ogrisel, if you didn't already know, conda-forge is basically replacing conda-recipes and anaconda-recipes. Some audited selection of packages built by conda-forge will move into the `defaults` selection. The main bullet points are summarized [here]( https://github.com/conda-forge/conda-forge.github.io/issues/22#issuecomment-197117472 ). More details of this process are discussed in that thread.

I expect some packages like `mkl` linked `scikit-learn` and the like will remain under Continuum. Everything else will move here though. I do plan on at least having OpenBLAS support for packages here. In the future, we could look at alternatives Blis. I'll reach out to you and others once I have a more concrete proposal.

**Comment by rgbkrk** [comment]
This is super awesome, I'm really happy about conda-forge in general. Thank you for the ping @jakirkham. I'm happy to help maintain, though I'm way less involved in cloudpickle than I was before.

**Comment by ogrisel** [comment]
> I expect some packages like mkl linked scikit-learn and the like will remain under Continuum. Everything else will move here though. I do plan on at least having OpenBLAS support for packages here. In the future, we could look at alternatives Blis. I'll reach out to you and others once I have a more concrete proposal.

Are you subscribed to the [wheel-builders mailing list](https://mail.python.org/mailman/listinfo/wheel-builders)? We could probably mutualize some effort when you do so. AFAIK, the plan for numpy / scipy wheels is to start with atlas in the short term has it was found very robust on a wide variety of hardware and operating systems although significantly slower than OpenBLAS. Once we have a build infra for an Atlas based stack, we will probably test OpenBLAS and Blis more extensively and maybe make it possible to have an install time or even runtime selection of the BLAS/LAPACK lib (via some kind of numpy import hook based on ctypes).

**Comment by jakirkham** [comment]
> This is super awesome, I'm really happy about conda-forge in general.

Me too. :smile:

> Thank you for the ping...

Of course, @rgbkrk.

> I'm happy to help maintain, though I'm way less involved in cloudpickle than I was before.

Cool, I'll add you to the feedstock.

**Comment by jakirkham** [comment]
> Are you subscribed to the wheel-builders mailing list?

No, but I will add myself. Thanks for the link, @ogrisel.

> We could probably mutualize some effort when you do so.

Definitely. I just have to iron out some details with features in conda first. Though this should have no effect on using them in wheels.

> AFAIK, the plan for numpy / scipy wheels is to start with atlas in the short term...

Yes, we should support ATLAS too. I just have OpenBLAS nearly ready to go so I figured I might start with that. Given that ATLAS is very valuable to the community at large, I would definitely appreciate as many eyes as possible on our implementation (once ready) to make sure we get this right.

> Once we have a build infra for an Atlas based stack, we will probably test OpenBLAS and Blis more extensively and maybe make it possible to have an install time or even runtime selection of the BLAS/LAPACK lib (via some kind of numpy import hook based on ctypes).

This proposal does sound very interesting. We will have to discuss more once we have decent builds for the different BLASes. Have you already talked to NumPy folks about this? Any relevant links would be nice so I can get up to speed.

**Pull request merged**